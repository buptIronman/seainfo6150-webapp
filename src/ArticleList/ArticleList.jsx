import React from 'react'
import ArticleListItem from "../ArticleListItem/ArticleListItem";

const ArticleList = (articles) => {
    console.log(articles.articles)
    if(!articles.articles.length){
        return (<div>You have no data!</div>)
    }

    return (<ul>{articles.articles.map((article) => {
        return <li>{ArticleListItem(article)}</li>
    })}</ul>)
}

export default ArticleList