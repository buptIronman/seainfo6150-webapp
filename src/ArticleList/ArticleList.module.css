.List{
    margin: 4px;
    background-color:  grey;
    border: #444444;
    border-style: dotted;
    border-radius: 15px;
    box-shadow: 5px 5px 10px 10px red;
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    column-count: 3;
}

ul{
    list-style-type: none;
}

@media screen and (min-width: 900px){
    .List{
        margin: 4px;
        background-color:  grey;
        border: #444444;
        border-style: dotted;
        border-radius: 15px;
        box-shadow: 5px 5px 10px 10px red;
        display: grid;
        grid-template-columns: 1fr 1fr 1fr;
        column-count: 3;
    }
}