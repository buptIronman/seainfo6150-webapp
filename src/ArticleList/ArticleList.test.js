import React from 'react';
import ArticleList from "../ArticleList/ArticleList";
import { fireEvent } from "@testing-library/react";
import AddingMachineHooks from "../component-examples/AddingMachine/AddingMachineHooks";

describe("ArticleList test", () => {
    it("no data", () => {
        const { container } = render(<ArticleList articles={0} />);
        expect(container).toMatchSnapshot();
    });

    it("renders correctly when currentNum is 10", () => {
        const { container } = render(<AddingMachineHooks startNum={10} />);
        expect(container).toMatchSnapshot();
    });

    it("renders correctly after button is clicked", () => {
        const { container, getByText } = render(
            <AddingMachineHooks startNum={0} />
        );
        const button = getByText("Click me to add 1");
        fireEvent.click(button);
        expect(container).toMatchSnapshot();
    });
})