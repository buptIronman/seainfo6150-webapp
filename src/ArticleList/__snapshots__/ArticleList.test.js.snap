// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList test no data 1`] = `
<div>
  <div
    class="List"
  >
    You have no data!
  </div>
</div>
`;

exports[`ArticleList test renders correctly after button is clicked 1`] = `
<div>
  <div
    style="border: 1px solid black; margin: 2rem;"
  >
    <header />
    <div>
      We started with 
      0
    </div>
    <div>
      We're now at 
      1
    </div>
    <button>
      Click me to add 1
    </button>
  </div>
</div>
`;

exports[`ArticleList test renders correctly when currentNum is 10 1`] = `
<div>
  <div
    style="border: 1px solid black; margin: 2rem;"
  >
    <header />
    <div>
      We started with 
      10
    </div>
    <div>
      We're now at 
      10
    </div>
    <button>
      Click me to add 1
    </button>
    <div>
      Whee!
    </div>
  </div>
</div>
`;
